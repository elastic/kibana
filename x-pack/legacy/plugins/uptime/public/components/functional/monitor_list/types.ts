/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */

export interface CondensedCheck {
  childStatuses: CondensedCheckStatus[];
  location: string | null;
  status: string;
  timestamp: string;
}

export interface CondensedCheckStatus {
  ip?: string | null;
  status: string;
  timestamp: string;
}

export interface Criteria {
  page?: {
    index: number;
    size: number;
  };
  sort?: {
    field: string;
    direction: 'asc' | 'desc';
  };
}

export interface ExpandedRowMap {
  [key: string]: JSX.Element;
}

export interface Pagination {
  hidePerPageOptions?: boolean;
  initialPageSize: number;
  pageIndex: number;
  pageSize: number;
  pageSizeOptions: number[];
  totalItemCount: number;
}
