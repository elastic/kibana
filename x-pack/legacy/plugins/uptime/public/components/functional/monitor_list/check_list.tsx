/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */

import React, { Fragment } from 'react';
import { EuiFlexGrid, EuiFlexItem, EuiText } from '@elastic/eui';
import { get } from 'lodash';
import { MonitorListStatusColumn } from './monitor_list_status_column';
import { Check } from '../../../../common/graphql/types';
import { LocationLink } from './location_link';

interface CheckListProps {
  checks: Check[];
}

export const CheckList = ({ checks }: CheckListProps) => {
  return (
    <EuiFlexGrid columns={3} gutterSize="l" style={{ paddingLeft: '40px' }}>
      {checks.map(check => {
        const location = get<string | null>(check, 'observer.geo.name', null);
        const agentId = get<string>(check, 'agent.id', 'null');
        const key = location + agentId + check.monitor.ip;
        return (
          <Fragment key={key}>
            <EuiFlexItem grow={4}>
              <MonitorListStatusColumn status={check.monitor.status} timestamp={check.timestamp} />
            </EuiFlexItem>
            <EuiFlexItem grow={4}>
              <LocationLink location={location} />
            </EuiFlexItem>
            <EuiFlexItem grow={4}>
              <EuiText color="secondary" size="s">
                {check.monitor.ip}
              </EuiText>
            </EuiFlexItem>
          </Fragment>
        );
      })}
    </EuiFlexGrid>
  );
};
