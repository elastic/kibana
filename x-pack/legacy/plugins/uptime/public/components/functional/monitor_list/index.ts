/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */

export { MonitorList } from './monitor_list';
export { Criteria, Pagination } from './types';
export { LocationLink } from './monitor_list_drawer';
