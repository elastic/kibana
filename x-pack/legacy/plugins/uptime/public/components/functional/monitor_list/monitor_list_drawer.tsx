/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */

import React from 'react';
import { MonitorSummary } from '../../../../common/graphql/types';
import { CheckList } from './check_list';
import { toCondensedCheck } from './to_condensed_check';
import { CondensedCheckList } from './condensed_check_list';

interface MonitorListDrawerProps {
  summary?: MonitorSummary;
  dangerColor: string;
  successColor: string;
  /**
   * The number of checks the component should fully render
   * before squashing them to single rows with condensed details.
   */
  condensedCheckLimit: number;
}

/**
 * The elements shown when the user expands the monitor list rows.
 */
export const MonitorListDrawer = ({
  condensedCheckLimit,
  dangerColor,
  successColor,
  summary,
}: MonitorListDrawerProps) => {
  if (!summary || !summary.state.checks) {
    return null;
  }
  if (summary.state.checks.length < condensedCheckLimit) {
    return <CheckList checks={summary.state.checks} />;
  } else {
    return (
      <CondensedCheckList
        condensedChecks={toCondensedCheck(summary.state.checks)}
        dangerColor={dangerColor}
        successColor={successColor}
      />
    );
  }
};
