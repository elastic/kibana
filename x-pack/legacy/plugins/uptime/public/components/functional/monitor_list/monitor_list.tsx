/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */

import { EuiBasicTable, EuiPanel, EuiTitle, EuiButtonIcon, EuiIcon, EuiLink } from '@elastic/eui';
import { EuiSpacer } from '@elastic/eui';
import { FormattedMessage } from '@kbn/i18n/react';
import { i18n } from '@kbn/i18n';
import { get } from 'lodash';
import React, { useState, Fragment } from 'react';
import { withUptimeGraphQL, UptimeGraphQLQueryProps } from '../../higher_order';
import { monitorStatesQuery } from '../../../queries/monitor_states_query';
import {
  MonitorSummary,
  MonitorSummaryResult,
  SummaryHistogramPoint,
} from '../../../../common/graphql/types';
import { MonitorListStatusColumn } from './monitor_list_status_column';
import { formatUptimeGraphQLErrorList } from '../../../lib/helper/format_error_list';
import { ExpandedRowMap } from './types';
import { MonitorListDrawer } from './monitor_list_drawer';
import { CLIENT_DEFAULTS } from '../../../../common/constants';
import { MonitorBarSeries } from '../charts';
import { MonitorPageLink } from '../monitor_page_link';
import { MonitorListActionsPopover } from './monitor_list_actions_popover';

interface MonitorListQueryResult {
  monitorStates?: MonitorSummaryResult;
}

interface MonitorListProps {
  absoluteStartDate: number;
  absoluteEndDate: number;
  dangerColor: string;
  successColor: string;
  linkParameters?: string;
  // TODO: reintegrate pagination in a future release
  // pageIndex: number;
  // pageSize: number;
  // TODO: reintroduce for pagination and sorting
  // onChange: (criteria: Criteria) => void;
  // TODO: reintegrate sorting in a future release
  // sortField: string;
  // sortDirection: string;
}

type Props = UptimeGraphQLQueryProps<MonitorListQueryResult> & MonitorListProps;

export const MonitorListComponent = (props: Props) => {
  const {
    absoluteStartDate,
    absoluteEndDate,
    dangerColor,
    successColor,
    data,
    errors,
    linkParameters,
    loading,
    // TODO: reintroduce for pagination and sorting
    // onChange,
    // TODO: reintegrate pagination in future release
    // pageIndex,
    // pageSize,
    // TODO: reintegrate sorting in future release
    // sortDirection,
    // sortField,
  } = props;
  const [drawerIds, updateDrawerIds] = useState<string[]>([]);

  const items = get<MonitorSummary[]>(data, 'monitorStates.summaries', []);
  // TODO: use with pagination
  // const count = get<number>(data, 'monitorStates.totalSummaryCount.count', 0);

  // TODO: reintegrate pagination in future release
  // const pagination: Pagination = {
  //   pageIndex,
  //   pageSize,
  //   pageSizeOptions: [5, 10, 20, 50],
  //   totalItemCount: count,
  //   hidePerPageOptions: false,
  // };

  // TODO: reintegrate sorting in future release
  // const sorting = {
  //   sort: {
  //     field: sortField,
  //     direction: sortDirection,
  //   },
  // };

  return (
    <Fragment>
      <EuiPanel>
        <EuiTitle size="xs">
          <h5>
            <FormattedMessage
              id="xpack.uptime.monitorList.monitoringStatusTitle"
              defaultMessage="Monitor status"
            />
          </h5>
        </EuiTitle>
        <EuiSpacer size="s" />
        <EuiBasicTable
          error={errors ? formatUptimeGraphQLErrorList(errors) : errors}
          loading={loading}
          isExpandable={true}
          hasActions={true}
          itemId="monitor_id"
          itemIdToExpandedRowMap={drawerIds.reduce((map: ExpandedRowMap, id: string) => {
            return {
              ...map,
              [id]: (
                <MonitorListDrawer
                  condensedCheckLimit={CLIENT_DEFAULTS.CONDENSED_CHECK_LIMIT}
                  summary={
                    items ? items.find(({ monitor_id: monitorId }) => monitorId === id) : undefined
                  }
                  successColor={successColor}
                  dangerColor={dangerColor}
                />
              ),
            };
          }, {})}
          items={items}
          // TODO: not needed without sorting and pagination
          // onChange={onChange}
          noItemsMessage={i18n.translate('xpack.uptime.monitorList.noItemMessage', {
            defaultMessage: 'No uptime monitors found',
            description:
              'This message is shown if the monitors table is rendered but has no items.',
          })}
          // TODO: reintegrate pagination in future release
          // pagination={pagination}
          // TODO: reintegrate sorting in future release
          // sorting={sorting}
          columns={[
            {
              align: 'left',
              field: 'state.monitor.status',
              name: i18n.translate('xpack.uptime.monitorList.statusColumnLabel', {
                defaultMessage: 'Status',
              }),
              render: (status: string, { state: { timestamp } }: MonitorSummary) => {
                return <MonitorListStatusColumn status={status} timestamp={timestamp} />;
              },
            },
            {
              align: 'left',
              field: 'state.monitor.name',
              name: i18n.translate('xpack.uptime.monitorList.nameColumnLabel', {
                defaultMessage: 'Name',
              }),
              render: (name: string, summary: MonitorSummary) => (
                <MonitorPageLink
                  id={summary.monitor_id}
                  linkParameters={linkParameters}
                  location={undefined}
                >
                  {name ? name : `Unnamed - ${summary.monitor_id}`}
                </MonitorPageLink>
              ),
              sortable: true,
            },
            {
              align: 'left',
              field: 'state.url.full',
              name: i18n.translate('xpack.uptime.monitorList.urlColumnLabel', {
                defaultMessage: 'URL',
              }),
              render: (url: string, summary: MonitorSummary) => (
                <Fragment>
                  <EuiLink href={url} target="_blank" color="text">
                    {url} <EuiIcon size="s" type="popout" color="subbdued" />
                  </EuiLink>
                </Fragment>
              ),
              sortable: true,
            },
            {
              field: 'histogram.points',
              name: i18n.translate('xpack.uptime.monitorList.monitorHistoryColumnLabel', {
                defaultMessage: 'Downtime history',
              }),
              mobileOptions: {
                show: false,
              },
              render: (histogramSeries: SummaryHistogramPoint[] | null) => (
                <MonitorBarSeries
                  absoluteStartDate={absoluteStartDate}
                  absoluteEndDate={absoluteEndDate}
                  dangerColor={dangerColor}
                  histogramSeries={histogramSeries}
                />
              ),
            },
            {
              id: 'actions',
              align: 'right',
              field: 'state',
              hasActions: true,
              mobileOptions: {
                header: false,
              },
              name: i18n.translate(
                'xpack.uptime.monitorList.observabilityIntegrationsColumnLabel',
                {
                  defaultMessage: 'Integrations',
                  description:
                    'The heading column of some action buttons that will take users to other Obsevability apps',
                }
              ),
              render: (state: any, summary: MonitorSummary) => (
                <MonitorListActionsPopover summary={summary} />
              ),
            },
            {
              align: 'left',
              field: 'monitor_id',
              name: '',
              sortable: true,
              width: '40px',
              isExpander: true,
              render: (id: string) => {
                return (
                  <EuiButtonIcon
                    aria-label={i18n.translate(
                      'xpack.uptime.monitorList.expandDrawerButton.ariaLabel',
                      {
                        defaultMessage: 'Expand row for monitor with ID {id}',
                        description:
                          'The user can click a button on this table and expand further details.',
                        values: {
                          id,
                        },
                      }
                    )}
                    iconType={drawerIds.find(item => item === id) ? 'arrowUp' : 'arrowDown'}
                    onClick={() => {
                      if (drawerIds.find(i => id === i)) {
                        updateDrawerIds(drawerIds.filter(p => p !== id));
                      } else {
                        updateDrawerIds([...drawerIds, id]);
                      }
                    }}
                  />
                );
              },
            },
          ]}
        />
      </EuiPanel>
    </Fragment>
  );
};

export const MonitorList = withUptimeGraphQL<MonitorListQueryResult, MonitorListProps>(
  MonitorListComponent,
  monitorStatesQuery
);
