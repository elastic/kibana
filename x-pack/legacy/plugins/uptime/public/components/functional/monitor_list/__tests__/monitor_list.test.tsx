/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */

import { shallowWithIntl } from 'test_utils/enzyme_helpers';
import React from 'react';
import { MonitorSummaryResult } from '../../../../../common/graphql/types';
import { MonitorListComponent } from '../monitor_list';

describe('MonitorList component', () => {
  let result: MonitorSummaryResult;

  beforeEach(() => {
    result = {
      summaries: [
        {
          monitor_id: 'foo',
          state: {
            checks: [
              {
                monitor: {
                  ip: '127.0.0.1',
                  status: 'up',
                },
                timestamp: '124',
              },
              {
                monitor: {
                  ip: '127.0.0.2',
                  status: 'down',
                },
                timestamp: '125',
              },
              {
                monitor: {
                  ip: '127.0.0.3',
                  status: 'down',
                },
                timestamp: '126',
              },
            ],
            summary: {
              up: 1,
              down: 2,
            },
            timestamp: '123',
          },
        },
        {
          monitor_id: 'bar',
          state: {
            checks: [
              {
                monitor: {
                  ip: '127.0.0.1',
                  status: 'up',
                },
                timestamp: '125',
              },
              {
                monitor: {
                  ip: '127.0.0.2',
                  status: 'up',
                },
                timestamp: '126',
              },
            ],
            summary: {
              up: 2,
              down: 0,
            },
            timestamp: '125',
          },
        },
      ],
      totalSummaryCount: {
        count: 2,
      },
    };
  });

  it('renders the monitor list', () => {
    const component = shallowWithIntl(
      <MonitorListComponent
        absoluteStartDate={123}
        absoluteEndDate={125}
        dangerColor="danger"
        data={{ monitorStates: result }}
        hasActiveFilters={false}
        loading={false}
        successColor="primary"
      />
    );

    expect(component).toMatchSnapshot();
  });

  it('renders a no items message when no data is provided', () => {
    const component = shallowWithIntl(
      <MonitorListComponent
        absoluteStartDate={123}
        absoluteEndDate={125}
        dangerColor="danger"
        data={{}}
        hasActiveFilters={false}
        loading={false}
        successColor="primary"
      />
    );
    expect(component).toMatchSnapshot();
  });
});
