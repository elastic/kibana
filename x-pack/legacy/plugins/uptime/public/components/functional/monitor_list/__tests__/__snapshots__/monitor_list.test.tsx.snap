// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MonitorList component renders a no items message when no data is provided 1`] = `
<Fragment>
  <EuiPanel>
    <EuiTitle
      size="xs"
    >
      <h5>
        <FormattedMessage
          defaultMessage="Monitor status"
          id="xpack.uptime.monitorList.monitoringStatusTitle"
          values={Object {}}
        />
      </h5>
    </EuiTitle>
    <EuiSpacer
      size="s"
    />
    <EuiBasicTable
      aria-label="Monitor Status table with columns for Status, Name, URL, IP, Downtime History and Integrations. The table is currently displaying 0 items."
      columns={
        Array [
          Object {
            "align": "left",
            "field": "state.monitor.status",
            "mobileOptions": Object {
              "fullWidth": true,
            },
            "name": "Status",
            "render": [Function],
          },
          Object {
            "align": "left",
            "field": "state.monitor.name",
            "mobileOptions": Object {
              "fullWidth": true,
            },
            "name": "Name",
            "render": [Function],
            "sortable": true,
          },
          Object {
            "align": "left",
            "field": "state.url.full",
            "name": "Url",
            "render": [Function],
          },
          Object {
            "align": "center",
            "field": "histogram.points",
            "mobileOptions": Object {
              "show": false,
            },
            "name": "Downtime history",
            "render": [Function],
          },
          Object {
            "align": "right",
            "field": "monitor_id",
            "isExpander": true,
            "name": "",
            "render": [Function],
            "sortable": true,
            "width": "24px",
          },
        ]
      }
      hasActions={true}
      isExpandable={true}
      itemId="monitor_id"
      itemIdToExpandedRowMap={Object {}}
      items={Array []}
      loading={false}
      noItemsMessage="No uptime monitors found"
      responsive={true}
      tableLayout="fixed"
    />
    <EuiSpacer
      size="m"
    />
    <EuiFlexGroup
      responsive={false}
    >
      <EuiFlexItem
        grow={false}
      >
        <OverviewPageLink
          dataTestSubj="xpack.uptime.monitorList.prevButton"
          direction="prev"
          pagination=""
        />
      </EuiFlexItem>
      <EuiFlexItem
        grow={false}
      >
        <OverviewPageLink
          dataTestSubj="xpack.uptime.monitorList.nextButton"
          direction="next"
          pagination=""
        />
      </EuiFlexItem>
    </EuiFlexGroup>
  </EuiPanel>
</Fragment>
`;

exports[`MonitorList component renders the monitor list 1`] = `
.c1 {
  padding-left: 17px;
}

.c2 {
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}

@media (max-width:574px) {
  .c0 {
    min-width: 230px;
  }
}

<div
  class="euiPanel euiPanel--paddingMedium"
>
  <h5
    class="euiTitle euiTitle--xsmall"
  >
    Monitor status
  </h5>
  <div
    class="euiSpacer euiSpacer--s"
  />
  <div
    aria-label="Monitor Status table with columns for Status, Name, URL, IP, Downtime History and Integrations. The table is currently displaying 2 items."
    class="euiBasicTable"
  >
    <div>
      <div
        class="euiTableHeaderMobile"
      >
        <div
          class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--alignItemsBaseline euiFlexGroup--justifyContentSpaceBetween euiFlexGroup--directionRow"
        >
          <div
            class="euiFlexItem euiFlexItem--flexGrowZero"
          />
          <div
            class="euiFlexItem euiFlexItem--flexGrowZero"
          />
        </div>
      </div>
      <table
        class="euiTable euiTable--responsive"
      >
        <caption
          aria-live="polite"
          aria-relevant="text"
          class="euiScreenReaderOnly euiTableCaption"
          role="status"
        />
        <thead>
          <tr>
            <th
              class="euiTableHeaderCell euiTableHeaderCell--hideForMobile"
              data-test-subj="tableHeaderCell_state.monitor.status_0"
              role="columnheader"
              scope="col"
            >
              <div
                class="euiTableCellContent"
              >
                <span
                  class="euiTableCellContent__text"
                >
                  Status
                </span>
              </div>
            </th>
            <th
              class="euiTableHeaderCell euiTableHeaderCell--hideForMobile"
              data-test-subj="tableHeaderCell_state.monitor.name_1"
              role="columnheader"
              scope="col"
            >
              <div
                class="euiTableCellContent"
              >
                <span
                  class="euiTableCellContent__text"
                >
                  Name
                </span>
              </div>
            </th>
            <th
              class="euiTableHeaderCell"
              data-test-subj="tableHeaderCell_state.url.full_2"
              role="columnheader"
              scope="col"
            >
              <div
                class="euiTableCellContent"
              >
                <span
                  class="euiTableCellContent__text"
                >
                  Url
                </span>
              </div>
            </th>
            <th
              class="euiTableHeaderCell euiTableHeaderCell--hideForMobile"
              data-test-subj="tableHeaderCell_histogram.points_3"
              role="columnheader"
              scope="col"
            >
              <div
                class="euiTableCellContent euiTableCellContent--alignCenter"
              >
                <span
                  class="euiTableCellContent__text"
                >
                  Downtime history
                </span>
              </div>
            </th>
            <th
              class="euiTableHeaderCell"
              data-test-subj="tableHeaderCell_monitor_id_4"
              role="columnheader"
              scope="col"
              style="width:24px"
            >
              <div
                class="euiTableCellContent euiTableCellContent--alignRight"
              >
                <span
                  class="euiTableCellContent__text"
                />
              </div>
            </th>
          </tr>
        </thead>
        <tbody>
          <tr
            class="euiTableRow euiTableRow-hasActions euiTableRow-isExpandable"
          >
            <td
              class="euiTableRowCell euiTableRowCell--isMobileFullWidth"
            >
              <div
                class="euiTableRowCell__mobileHeader euiTableRowCell--hideForDesktop"
              >
                Status
              </div>
              <div
                class="euiTableCellContent euiTableCellContent--overflowingContent"
              >
                <div
                  class="euiFlexGroup euiFlexGroup--alignItemsCenter euiFlexGroup--directionRow c0"
                >
                  <div
                    class="euiFlexItem euiFlexItem--flexGrow1"
                    style="flex-basis:40px"
                  >
                    <div
                      class="euiHealth"
                      style="display:block"
                    >
                      <div
                        class="euiFlexGroup euiFlexGroup--gutterExtraSmall euiFlexGroup--alignItemsCenter euiFlexGroup--directionRow"
                      >
                        <div
                          class="euiFlexItem euiFlexItem--flexGrowZero"
                        >
                          <svg
                            aria-hidden="true"
                            class="euiIcon euiIcon--medium euiIcon-isLoading"
                            focusable="false"
                            height="16"
                            role="img"
                            viewBox="0 0 16 16"
                            width="16"
                            xmlns="http://www.w3.org/2000/svg"
                          />
                        </div>
                        <div
                          class="euiFlexItem euiFlexItem--flexGrowZero"
                        />
                      </div>
                    </div>
                    <span
                      class="c1"
                    >
                      <span
                        class="euiToolTipAnchor"
                      >
                        <div
                          class="euiText euiText--extraSmall"
                        >
                          <div
                            class="euiTextColor euiTextColor--subdued"
                          >
                            1897 Yr ago
                          </div>
                        </div>
                      </span>
                    </span>
                  </div>
                  <div
                    class="euiFlexItem euiFlexItem--flexGrow2"
                  >
                    <div
                      class="euiText euiText--small"
                    >
                      in 0/1 Location
                    </div>
                  </div>
                </div>
              </div>
            </td>
            <td
              class="euiTableRowCell euiTableRowCell--isMobileFullWidth"
            >
              <div
                class="euiTableRowCell__mobileHeader euiTableRowCell--hideForDesktop"
              >
                Name
              </div>
              <div
                class="euiTableCellContent euiTableCellContent--overflowingContent"
              >
                <button
                  class="euiLink euiLink--primary"
                  type="button"
                >
                  <a
                    data-test-subj="monitor-page-link-foo"
                    href="/monitor/Zm9v"
                  >
                    Unnamed - foo
                  </a>
                </button>
              </div>
            </td>
            <td
              class="euiTableRowCell"
            >
              <div
                class="euiTableRowCell__mobileHeader euiTableRowCell--hideForDesktop"
              >
                Url
              </div>
              <div
                class="euiTableCellContent euiTableCellContent--overflowingContent"
              >
                <button
                  class="euiLink euiLink--text c2"
                  type="button"
                >
                   
                  <svg
                    aria-hidden="true"
                    class="euiIcon euiIcon--small euiIcon-isLoading"
                    focusable="false"
                    height="16"
                    role="img"
                    style="fill:subbdued"
                    viewBox="0 0 16 16"
                    width="16"
                    xmlns="http://www.w3.org/2000/svg"
                  />
                </button>
              </div>
            </td>
            <td
              class="euiTableRowCell euiTableRowCell--hideForMobile"
            >
              <div
                class="euiTableCellContent euiTableCellContent--alignCenter euiTableCellContent--overflowingContent"
              >
                <span
                  class="euiToolTipAnchor"
                >
                  <div
                    class="euiText euiText--medium"
                  >
                    <div
                      class="euiTextColor euiTextColor--secondary"
                    >
                      --
                    </div>
                  </div>
                </span>
              </div>
            </td>
            <td
              class="euiTableRowCell euiTableRowCell--isExpander"
              style="width:24px"
            >
              <div
                class="euiTableCellContent euiTableCellContent--alignRight euiTableCellContent--overflowingContent"
              >
                <button
                  aria-label="Expand row for monitor with ID foo"
                  class="euiButtonIcon euiButtonIcon--primary"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="euiIcon euiIcon--medium euiIcon-isLoading euiButtonIcon__icon"
                    focusable="false"
                    height="16"
                    role="img"
                    viewBox="0 0 16 16"
                    width="16"
                    xmlns="http://www.w3.org/2000/svg"
                  />
                </button>
              </div>
            </td>
          </tr>
          <tr
            class="euiTableRow euiTableRow-hasActions euiTableRow-isExpandable"
          >
            <td
              class="euiTableRowCell euiTableRowCell--isMobileFullWidth"
            >
              <div
                class="euiTableRowCell__mobileHeader euiTableRowCell--hideForDesktop"
              >
                Status
              </div>
              <div
                class="euiTableCellContent euiTableCellContent--overflowingContent"
              >
                <div
                  class="euiFlexGroup euiFlexGroup--alignItemsCenter euiFlexGroup--directionRow c0"
                >
                  <div
                    class="euiFlexItem euiFlexItem--flexGrow1"
                    style="flex-basis:40px"
                  >
                    <div
                      class="euiHealth"
                      style="display:block"
                    >
                      <div
                        class="euiFlexGroup euiFlexGroup--gutterExtraSmall euiFlexGroup--alignItemsCenter euiFlexGroup--directionRow"
                      >
                        <div
                          class="euiFlexItem euiFlexItem--flexGrowZero"
                        >
                          <svg
                            aria-hidden="true"
                            class="euiIcon euiIcon--medium euiIcon-isLoading"
                            focusable="false"
                            height="16"
                            role="img"
                            viewBox="0 0 16 16"
                            width="16"
                            xmlns="http://www.w3.org/2000/svg"
                          />
                        </div>
                        <div
                          class="euiFlexItem euiFlexItem--flexGrowZero"
                        />
                      </div>
                    </div>
                    <span
                      class="c1"
                    >
                      <span
                        class="euiToolTipAnchor"
                      >
                        <div
                          class="euiText euiText--extraSmall"
                        >
                          <div
                            class="euiTextColor euiTextColor--subdued"
                          >
                            1895 Yr ago
                          </div>
                        </div>
                      </span>
                    </span>
                  </div>
                  <div
                    class="euiFlexItem euiFlexItem--flexGrow2"
                  >
                    <div
                      class="euiText euiText--small"
                    >
                      in 1/1 Location
                    </div>
                  </div>
                </div>
              </div>
            </td>
            <td
              class="euiTableRowCell euiTableRowCell--isMobileFullWidth"
            >
              <div
                class="euiTableRowCell__mobileHeader euiTableRowCell--hideForDesktop"
              >
                Name
              </div>
              <div
                class="euiTableCellContent euiTableCellContent--overflowingContent"
              >
                <button
                  class="euiLink euiLink--primary"
                  type="button"
                >
                  <a
                    data-test-subj="monitor-page-link-bar"
                    href="/monitor/YmFy"
                  >
                    Unnamed - bar
                  </a>
                </button>
              </div>
            </td>
            <td
              class="euiTableRowCell"
            >
              <div
                class="euiTableRowCell__mobileHeader euiTableRowCell--hideForDesktop"
              >
                Url
              </div>
              <div
                class="euiTableCellContent euiTableCellContent--overflowingContent"
              >
                <button
                  class="euiLink euiLink--text c2"
                  type="button"
                >
                   
                  <svg
                    aria-hidden="true"
                    class="euiIcon euiIcon--small euiIcon-isLoading"
                    focusable="false"
                    height="16"
                    role="img"
                    style="fill:subbdued"
                    viewBox="0 0 16 16"
                    width="16"
                    xmlns="http://www.w3.org/2000/svg"
                  />
                </button>
              </div>
            </td>
            <td
              class="euiTableRowCell euiTableRowCell--hideForMobile"
            >
              <div
                class="euiTableCellContent euiTableCellContent--alignCenter euiTableCellContent--overflowingContent"
              >
                <span
                  class="euiToolTipAnchor"
                >
                  <div
                    class="euiText euiText--medium"
                  >
                    <div
                      class="euiTextColor euiTextColor--secondary"
                    >
                      --
                    </div>
                  </div>
                </span>
              </div>
            </td>
            <td
              class="euiTableRowCell euiTableRowCell--isExpander"
              style="width:24px"
            >
              <div
                class="euiTableCellContent euiTableCellContent--alignRight euiTableCellContent--overflowingContent"
              >
                <button
                  aria-label="Expand row for monitor with ID bar"
                  class="euiButtonIcon euiButtonIcon--primary"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="euiIcon euiIcon--medium euiIcon-isLoading euiButtonIcon__icon"
                    focusable="false"
                    height="16"
                    role="img"
                    viewBox="0 0 16 16"
                    width="16"
                    xmlns="http://www.w3.org/2000/svg"
                  />
                </button>
              </div>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
  <div
    class="euiSpacer euiSpacer--m"
  />
  <div
    class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--directionRow"
  >
    <div
      class="euiFlexItem euiFlexItem--flexGrowZero"
    >
      <button
        aria-label="A disabled pagination button indicating that there cannot be any further navigation in the monitors list."
        class="euiButtonIcon euiButtonIcon--text"
        data-test-subj="xpack.uptime.monitorList.prevButton"
        disabled=""
        type="button"
      >
        <svg
          aria-hidden="true"
          class="euiIcon euiIcon--medium euiIcon-isLoading euiButtonIcon__icon"
          focusable="false"
          height="16"
          role="img"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        />
      </button>
    </div>
    <div
      class="euiFlexItem euiFlexItem--flexGrowZero"
    >
      <button
        aria-label="A disabled pagination button indicating that there cannot be any further navigation in the monitors list."
        class="euiButtonIcon euiButtonIcon--text"
        data-test-subj="xpack.uptime.monitorList.nextButton"
        disabled=""
        type="button"
      >
        <svg
          aria-hidden="true"
          class="euiIcon euiIcon--medium euiIcon-isLoading euiButtonIcon__icon"
          focusable="false"
          height="16"
          role="img"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        />
      </button>
    </div>
  </div>
</div>
`;

exports[`MonitorList component shallow renders the monitor list 1`] = `
<Fragment>
  <EuiPanel>
    <EuiTitle
      size="xs"
    >
      <h5>
        <FormattedMessage
          defaultMessage="Monitor status"
          id="xpack.uptime.monitorList.monitoringStatusTitle"
          values={Object {}}
        />
      </h5>
    </EuiTitle>
    <EuiSpacer
      size="s"
    />
    <EuiBasicTable
      aria-label="Monitor Status table with columns for Status, Name, URL, IP, Downtime History and Integrations. The table is currently displaying 2 items."
      columns={
        Array [
          Object {
            "align": "left",
            "field": "state.monitor.status",
            "mobileOptions": Object {
              "fullWidth": true,
            },
            "name": "Status",
            "render": [Function],
          },
          Object {
            "align": "left",
            "field": "state.monitor.name",
            "mobileOptions": Object {
              "fullWidth": true,
            },
            "name": "Name",
            "render": [Function],
            "sortable": true,
          },
          Object {
            "align": "left",
            "field": "state.url.full",
            "name": "Url",
            "render": [Function],
          },
          Object {
            "align": "center",
            "field": "histogram.points",
            "mobileOptions": Object {
              "show": false,
            },
            "name": "Downtime history",
            "render": [Function],
          },
          Object {
            "align": "right",
            "field": "monitor_id",
            "isExpander": true,
            "name": "",
            "render": [Function],
            "sortable": true,
            "width": "24px",
          },
        ]
      }
      hasActions={true}
      isExpandable={true}
      itemId="monitor_id"
      itemIdToExpandedRowMap={Object {}}
      items={
        Array [
          Object {
            "monitor_id": "foo",
            "state": Object {
              "checks": Array [
                Object {
                  "monitor": Object {
                    "ip": "127.0.0.1",
                    "status": "up",
                  },
                  "timestamp": "124",
                },
                Object {
                  "monitor": Object {
                    "ip": "127.0.0.2",
                    "status": "down",
                  },
                  "timestamp": "125",
                },
                Object {
                  "monitor": Object {
                    "ip": "127.0.0.3",
                    "status": "down",
                  },
                  "timestamp": "126",
                },
              ],
              "summary": Object {
                "down": 2,
                "up": 1,
              },
              "timestamp": "123",
            },
          },
          Object {
            "monitor_id": "bar",
            "state": Object {
              "checks": Array [
                Object {
                  "monitor": Object {
                    "ip": "127.0.0.1",
                    "status": "up",
                  },
                  "timestamp": "125",
                },
                Object {
                  "monitor": Object {
                    "ip": "127.0.0.2",
                    "status": "up",
                  },
                  "timestamp": "126",
                },
              ],
              "summary": Object {
                "down": 0,
                "up": 2,
              },
              "timestamp": "125",
            },
          },
        ]
      }
      loading={false}
      noItemsMessage="No uptime monitors found"
      responsive={true}
      tableLayout="fixed"
    />
    <EuiSpacer
      size="m"
    />
    <EuiFlexGroup
      responsive={false}
    >
      <EuiFlexItem
        grow={false}
      >
        <OverviewPageLink
          dataTestSubj="xpack.uptime.monitorList.prevButton"
          direction="prev"
          pagination=""
        />
      </EuiFlexItem>
      <EuiFlexItem
        grow={false}
      >
        <OverviewPageLink
          dataTestSubj="xpack.uptime.monitorList.nextButton"
          direction="next"
          pagination=""
        />
      </EuiFlexItem>
    </EuiFlexGroup>
  </EuiPanel>
</Fragment>
`;
